### PR TITLE
change: add json validation for device schema in unit tests

### PR DESCRIPTION
**Pull request by math411**
***Issue** #, if available:*

*Description of changes:*
Add json schema validation for device schemas within the tests.

*Testing done:*
Unit test validation added. Integ tests and unit tests still run.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md) doc
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/ocean_plugin/conftest.py [hunk 1]
```diff
@@ -39,6 +40,11 @@ def shots():
 
 @pytest.fixture
 def service_properties():
+    return {"shotsRange": [0, 10]}
+
+
+@pytest.fixture
+def service_properties_list():
```
Comment (kshitijc) [inline test/unit_tests/braket/ocean_plugin/conftest.py:47]: Could you please provide some context on why this is needed ?

File: test/unit_tests/braket/ocean_plugin/conftest.py [hunk 2]
```diff
@@ -84,64 +90,64 @@ def advantage_device_parameters():
...
+        "action": {
+            "braket.ir.annealing.problem": {
+                "actionType": "braket.ir.annealing.problem",
+                "version": ["1"],
+            }
+        },
+        "deviceParameters": two_thousand_q_device_parameters,
+    }
+    validate(d_wave_device_capabilities_json, DwaveDeviceCapabilities.schema())
+    return DwaveDeviceCapabilities.parse_obj(d_wave_device_capabilities_json)
```
Comment (kshitijc) [inline test/unit_tests/braket/ocean_plugin/conftest.py:118]: We are validating a json blob which we constructed for tests. How do we know if this json represents actual data and not something which has been morphed to get the tests to pass ?

**Review by kshitijc — COMMENTED** [review]
The core issue that we were trying to address is that the parameters that are passed as a part of the API request to the service do not conform to the device parameters jsonschema specification that is provided by the device capabilities.
We still aren't validating that here. Also think about which other objects could use validation.

File: test/unit_tests/braket/ocean_plugin/conftest.py (hunk 1 above)
Comment (math411) [reply test/unit_tests/braket/ocean_plugin/conftest.py:47]: The input expected is the array input but the comparison done later is converted to the list.

**Pull request closed without merging**